src/libs/actions/Delegate.ts

```diff
@@ -0,0 +1,155 @@
+
+let delegatedAccess: DelegatedAccess;
+Onyx.connect({
+    key: ONYXKEYS.ACCOUNT,
+    callback: (val) => {
+        delegatedAccess = val?.delegatedAccess ?? {};
+    },
+});
+
+const KEYS_TO_PRESERVE_DELEGATE_ACCESS = [ONYXKEYS.NVP_TRY_FOCUS_MODE, ONYXKEYS.PREFERRED_THEME, ONYXKEYS.NVP_PREFERRED_LOCALE, ONYXKEYS.SESSION];
+
+function connect(email: string) {
+    if (!delegatedAccess?.delegators) {
+        return;
+    }
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                delegatedAccess: {
+                    delegators: delegatedAccess.delegators.map((delegator) => (delegator.email === email ? {...delegator, error: undefined} : delegator)),
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                delegatedAccess: {
+                    delegators: delegatedAccess.delegators.map((delegator) => (delegator.email === email ? {...delegator, error: undefined} : delegator)),
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                delegatedAccess: {
+                    delegators: delegatedAccess.delegators.map((delegator) => (delegator.email === email ? {...delegator, error: 'delegate.genericError'} : delegator)),
+                },
+            },
+        },
+    ];
+
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.CONNECT_AS_DELEGATE, {to: email}, {optimisticData, successData, failureData})
+        .then((response) => {
+            if (!response?.restrictedToken || !response?.encryptedAuthToken) {
+                Log.alert('[Delegate] No auth token returned while connecting as a delegate');
+                Onyx.update(failureData);
+                return;
+            }
+            return SequentialQueue.waitForIdle()
+                .then(() => Onyx.clear(KEYS_TO_PRESERVE_DELEGATE_ACCESS))
+                .then(() => {
+                    // Update authToken in Onyx and in our local variables so that API requests will use the new authToken
+                    updateSessionAuthTokens(response?.restrictedToken, response?.encryptedAuthToken);
+
+                    NetworkStore.setAuthToken(response?.restrictedToken ?? null);
+                    openApp();
+                });
+        })
+        .catch((error) => {
+            Log.alert('[Delegate] Error connecting as delegate', {error});
+            Onyx.update(failureData);
+        });
+}
+
+function disconnect() {
+    const optimisticData: OnyxUpdate[] = [
```

Comment:
Should we optimistically clear `delegatedAccess` here, as well as the errors? In the video I just posted about that other bug, you can see we flash a doubled list briefly

<img width="397" alt="image" src="https://github.com/user-attachments/assets/0c1d7b78-1ee8-4272-bd04-5527ebe97071">

Reply:
@dangrous this was not due to the optimistic values of `delegatedAccess`, but it was a bug in `menuItems()`.

Fixed it